catalog/spatial/csw/spatial-csw-endpoint/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/TestCswEndpoint.java

```diff
@@ -2039,6 +2048,46 @@ private void setUpMocksForProductRetrieval(boolean includeMimeType)
                 resourceResponse);
     }
 
+    private List<QueryResponse> getQRBatch(int batchSize, int total) {
+        Queue<Result> results = new ArrayDeque<>();
+        for (int i = 1; i <= total; i++) {
+            MetacardImpl metacard = new MetacardImpl();
+            metacard.setId(i + "");
+            results.add(new ResultImpl(metacard));
+        }
+
+        List<QueryResponse> queryResponses = new ArrayList<>();
+        while (!results.isEmpty()) {
+            List<Result> batchList = new ArrayList<>();
+            for (int i = 0; i < batchSize; i++) {
+                Result result = results.poll();
```

Comment:
I'm not sure how I feel about trying to recreate catalog framework logic instead of just mocking the responses verbatim. Something feels wrong with duplicating its' logic.